packages/fxa-auth-server/lib/routes/validators.js

```diff
@@ -450,7 +450,21 @@ module.exports.subscriptionsSubscriptionValidator = isA.object({
     .optional()
     .allow(null)
     .description(DESCRIPTIONS.promotionDuration),
-  promotion_end: isA.number().optional().allow(null),
+  promotion_end: isA
+    .number()
+    .optional()
+    .allow(null)
+    .description(DESCRIPTIONS.promotionEnd),
+  promotion_name: isA
+    .string()
+    .optional()
+    .allow(null)
+    .description(DESCRIPTIONS.promotionName),
+  promotion_percent_off: isA
+    .number()
+    .optional()
+    .allow(null)
+    .description(DESCRIPTIONS.promotionPercentOff),
```

Comment:
Does this need a `.min(0).max(100)`to make sure nobody tries creating a 110% off promotion, or are we validating that elsewhere?

Reply:
and/or do we need to use [`.integer()`](https://joi.dev/api/?v=17.13.0#numberinteger) to avoid decimals?